datafusion/proto/src/to_proto.rs

```diff
@@ -1188,7 +1188,25 @@ impl TryFrom<&ScalarValue> for protobuf::ScalarValue {
                 })
             }
 
-            datafusion::scalar::ScalarValue::Time64(v) => {
+            datafusion::scalar::ScalarValue::Time32Second(v) => {
+                create_proto_scalar(v, PrimitiveScalarType::Time64, |v| {
+                    Value::Time64Value(*v as i64 * 1_000_000_000)
+                })
+            }
+
+            datafusion::scalar::ScalarValue::Time32Millisecond(v) => {
+                create_proto_scalar(v, PrimitiveScalarType::Time64, |v| {
+                    Value::Time64Value(*v as i64 * 1_000_000)
+                })
+            }
+
+            datafusion::scalar::ScalarValue::Time64Microsecond(v) => {
+                create_proto_scalar(v, PrimitiveScalarType::Time64, |v| {
+                    Value::Time64Value(*v * 1_000)
+                })
+            }
+
+            datafusion::scalar::ScalarValue::Time64Nanosecond(v) => {
```

Comment:
I agree that this is the best way for now to avoid blowing the scope of the PR. I recommend we put a comment above the added code to indicate that this is necessary because the protos only know `Time64` (and always interpret this as nanosecond accuracy).